pkg/controller/bucketclass/bucketclass_controller.go

```diff
@@ -52,29 +54,29 @@ func Add(mgr manager.Manager) error {
 
 	// Watch for changes on resources to trigger reconcile
 	err = c.Watch(&source.Kind{Type: &nbv1.BucketClass{}}, &handler.EnqueueRequestForObject{},
-		bucketClassPredicate, &logEventsPredicate)
+		ignoreUnmatchedProvisioner(options.Namespace), bucketClassPredicate, &logEventsPredicate)
```

Comment:
I think something similar came up in our initial bucketclass discussion where it was suggested that this behaviour would be fine. For example, if an operator is running in singlenamespace mode and a CR is created in another, the operator won't even know that the resource was created.

The issue with touching the resource (when it is not clear which one should) is that which operator (among many co-existing) should touch? All of them? First of them?

Reply:
We can touch the status to of the object to reflect that a certain operator is ignoring the resource because of unmatched name however won't that cause some clutter?

Reply:
Are you referring to which operator updates the object?

Reply:
I don't think we need to indicate anything to the user. 
think of a case (like our dev cluster) where we have many namespaces running noobaa-operator. if you create a BC that reference one of them using the `noobaa-operator` label, it's expected that all other operators will ignore it